File: 2609/2609.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    2609.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/09/08 11:52:04 by jihoolee          #+#    #+#              #
#    Updated: 2022/09/08 11:55:05 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_gcd(a, b):
    if (b == 0):
        return a
    return get_gcd(b, a % b)

def get_lcm(a, b):
    gcd = get_gcd(a, b)

    return a * b // gcd

def main():
    a, b = map(int, input().split())
    print(get_gcd(a, b))
    print(get_lcm(a, b))

if (__name__ == '__main__'):
    main()

File: 2609/2609.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   2609.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/09/08 11:47:08 by jihoolee          #+#    #+#             */
/*   Updated: 2022/09/08 11:51:23 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>

int get_gcd(int a, int b) {
  if (b == 0)
    return a;
  return get_gcd(b, a % b);
}

int get_lcm(int a, int b) {
  int gcd = get_gcd(a, b);

  return a * b / gcd;
}

int main(void) {
  int a, b;

  std::cin >> a >> b;
  std::cout << get_gcd(a, b) << std::endl;
  std::cout << get_lcm(a, b) << std::endl;
  return 0;
}
